setup.py

```diff
@@ -38,78 +60,126 @@ def get_tag(self):
-            "envs/llvm/service/compiler_gym-llvm-service",
-            "envs/loop_tool/service/compiler_gym-loop_tool-service",
-            "envs/llvm/service/libLLVMPolly.so",
-            "envs/llvm/service/compute_observation",
-            "third_party/cbench/benchmarks.txt",
-            "third_party/cbench/cbench-v*/*",
-            "third_party/csmith/csmith/bin/csmith",
-            "third_party/csmith/csmith/include/csmith-2.3.0/*.h",
-            "third_party/inst2vec/*.pickle",
+
+class build(distutils.command.build.build):
+    def initialize_options(self):
+        distutils.command.build.build.initialize_options(self)
+        self.build_base = args.build_dir
+
+
+# Add files that should be excluded from the package.
+# The argument exclude_package_data of setuptools.setup(...)
+# does not work with py files. They have to be excluded here.
+excluded = [
+    str(Path(args.package_dir).joinpath("compiler_gym/envs/llvm/make_specs.py")),
+    str(Path(args.package_dir).joinpath("compiler_gym/bin/random_eval.py")),
+]
+
+
+class build_py(build_py_orig):
+    def find_package_modules(self, package, package_dir):
+        modules = super().find_package_modules(package, package_dir)
+        res = [
+            (pkg, mod, file)
+            for (pkg, mod, file) in modules
+            if not any(fnmatch.fnmatchcase(file, pat=pattern) for pattern in excluded)
         ]
-    },
-    install_requires=requirements,
-    include_package_data=True,
-    python_requires=">=3.6",
-    classifiers=[
-        "Development Status :: 3 - Alpha",
-        "Environment :: Console",
-        "Intended Audience :: Developers",
-        "Intended Audience :: Science/Research",
-        "License :: OSI Approved :: MIT License",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3",
-        "Topic :: Scientific/Engineering :: Artificial Intelligence",
-        "Topic :: Software Development :: Compilers",
-    ],
-    cmdclass={"bdist_wheel": bdist_wheel},
-    platforms=[distutils.util.get_platform()],
-    zip_safe=False,
-)
+        return res
+
+
+def wheel_name(**kwargs):
+    # create a fake distribution from arguments
+    dist = Distribution(attrs=kwargs)
+    # finalize bdist_wheel command
+    bdist_wheel_cmd = dist.get_command_obj("bdist_wheel")
+    bdist_wheel_cmd.ensure_finalized()
+    # assemble wheel file name
+    distname = bdist_wheel_cmd.wheel_dist_name
+    tag = "-".join(bdist_wheel_cmd.get_tag())
+    return f"{distname}-{tag}.whl"
+
+
+if args.get_wheel_filename:
+    file_name = wheel_name(
+        name="compiler_gym",
+        version=version,
+        cmdclass={"bdist_wheel": bdist_wheel},
+        platforms=[distutils.util.get_platform()],
+    )
+    sys.stdout.write(file_name)
+else:
```

Comment:
This new conditional logic makes the setup.py file quite complicated (at least to me, without being familiar with the intent). Could you add a comment giving a high-level overview of what all this code is for? Maybe at the top of the file, or wherever feels best

Reply:
I added a comment.
The goal here is to be able to query the script for its byproducts. `setuptools` does not have an out-of-the-box facility for that. Then you can use this result in the build when describing the output of the target.

Reply:
Ah I see, that's clever :)